src/mscorlib/shared/System/IO/PathInternal.Windows.cs

```diff
@@ -214,25 +208,29 @@ private unsafe static int GetRootLength(char* path, int pathLength)
                     // (e.g. to \\?\UNC\Server\Share or \\Server\Share\)
                     i = uncRootLength;
                     int n = 2; // Maximum separators to skip
-                    while (i < pathLength && (!IsDirectorySeparator(path[i]) || --n > 0)) i++;
+                    while (i < pathLength && (!IsDirectorySeparator(path[i]) || --n > 0))
+                        i++;
                 }
             }
             else if (pathLength >= volumeSeparatorLength && path[volumeSeparatorLength - 1] == VolumeSeparatorChar)
             {
                 // Path is at least longer than where we expect a colon, and has a colon (\\?\A:, A:)
                 // If the colon is followed by a directory separator, move past it
                 i = volumeSeparatorLength;
-                if (pathLength >= volumeSeparatorLength + 1 && IsDirectorySeparator(path[volumeSeparatorLength])) i++;
+                if (pathLength >= volumeSeparatorLength + 1 && IsDirectorySeparator(path[volumeSeparatorLength]))
+                    i++;
             }
             return i;
         }
 
-        private unsafe static bool StartsWithOrdinal(char* source, int sourceLength, string value)
+        private static bool StartsWithOrdinal(ReadOnlySpan<char> source, int sourceLength, string value)
```

Comment:
Does this need to take sourceLength anymore?  Now that source is a span, call sites could just Slice.  But also a quick look at the call sites suggest that they all just pass the equivalent of source.Length, so a separate length shouldn't be needed, regardless.